massbalancemachine/data_processing/create_glacier_grid.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import salem
+import pyproj
+import pandas as pd
+
+def create_glacier_grid(ds, years, glacier_indices, gdir, rgi_gl):
+    """_summary_
+
+    Args:
+        ds (xarray.Dataset): glacier attributes masked over its outline
+        years (list): years of the data
+        glacier_indices (np.array): indices of glacier mask in the OGGM grid
+        gdir (oggm directory): oggm glacier directory
+        rgi_gl (str): RGI Id of the glacier
+
+    Returns:
+        df_grid (pd.DataFrame): dataframe of glacier grid data, for each year
+    """
+    # Assuming the coordinate variables are named 'x' and 'y' in your dataset
+    x_coords = ds['x'].values
+    y_coords = ds['y'].values
+    
+    # Retrieve the x and y values using the glacier indices
+    glacier_x_vals = x_coords[glacier_indices[1]]
+    glacier_y_vals = y_coords[glacier_indices[0]]
+
+    # Convert glacier coordinates to latitude and longitude
+    # Transform stake coord to glacier system:
+    transf = pyproj.Transformer.from_proj(gdir.grid.proj,
+                                        salem.wgs84,
+                                        always_xy=True)
+    lon, lat = transf.transform(glacier_x_vals, glacier_y_vals)
+    
+    # Glacier mask as boolean array:
+    gl_mask_bool = ds['glacier_mask'].values.astype(bool)
+
+    # Create a DataFrame
+    data_grid = {
+        'RGIId': [rgi_gl] * len(ds.masked_elev.values[gl_mask_bool]),
+        'POINT_LAT': lat,
+        'POINT_LON': lon,
+        'aspect': ds.masked_aspect.values[gl_mask_bool],
+        'slope': ds.masked_slope.values[gl_mask_bool],
+        'topo': ds.masked_elev.values[gl_mask_bool],
+        'dis_from_border': ds.masked_dis.values[gl_mask_bool],
+    }
+
+    df_grid = pd.DataFrame(data_grid)
+
+    # Match to WGMS format:
+    df_grid['POINT_ID'] = np.arange(1, len(df_grid) + 1)
+    df_grid["PERIOD"] = "annual"
```

Comment:
There is no column PERIOD in the WGMS database, so this might be redundant?

Reply:
Yes, good point; I'll add it separately.